### PR TITLE
Adding a patch to wagyu for windows build

### PR DESCRIPTION
**Pull request by saumyaj3**
In order to build wagyu in DEBUG mode, a patch is created in wagyu recipe.

CURA-11444

### CODE REVIEW

File: recipes/mapbox-wagyu/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("mapbox-geometry/2.0.3")
+
+    def package_id(self):
+        self.info.clear()
+
+    def export_sources(self):
+        export_conandata_patches(self)
```
Comment (jellespijker) [inline recipes/mapbox-wagyu/all/conanfile.py:47]: add newline

File: recipes/mapbox-wagyu/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
```
Comment (jellespijker) [inline recipes/mapbox-wagyu/all/conanfile.py:60]: remove newline

File: recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch [hunk 3]
```diff
@@ -0,0 +1,35 @@
+Subject: [PATCH] patch for wagyu
+---
+Index: include/mapbox/geometry/wagyu/ring.hpp
```
Comment (jellespijker) [inline recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch]: remove IDE specific information

File: recipes/mapbox-wagyu/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
```
Comment (jellespijker) [inline recipes/mapbox-wagyu/all/conanfile.py:28]: your adding `__has_include` make sure the minimum compiler versions reflect the support for this. See  https://en.cppreference.com/w/c/compiler_support

File: recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ #include <sstream>
+ #include <vector>
+ 
+-#ifdef DEBUG
++#if defined(DEBUG) &&  __has_include(<execinfo.h>)
+ #include <execinfo.h>
+ #include <iostream>
+ #include <sstream>
+Index: include/mapbox/geometry/wagyu/ring_util.hpp
+IDEA additional info:
```
Comment (jellespijker) [inline recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch]: remove all IDE specific lines

**Review by jellespijker — CHANGES_REQUESTED** [review]
lgtm; few minor changes and a remark about min compiler supprt

File: recipes/mapbox-wagyu/all/conanfile.py (hunk 1 above)
Comment (saumyaj3) [reply recipes/mapbox-wagyu/all/conanfile.py:47]: d8ac36bf52838cd9d54254f6987b4262e44d8640

File: recipes/mapbox-wagyu/all/conanfile.py (hunk 2 above)
Comment (saumyaj3) [reply recipes/mapbox-wagyu/all/conanfile.py:60]: d8ac36bf52838cd9d54254f6987b4262e44d8640

File: recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch (hunk 3 above)
Comment (saumyaj3) [reply recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch]: d8ac36bf52838cd9d54254f6987b4262e44d8640

File: recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch (hunk 5 above)
Comment (saumyaj3) [reply recipes/mapbox-wagyu/all/patches/0001-patch-for-execinfo-file.patch]: d8ac36bf52838cd9d54254f6987b4262e44d8640

File: recipes/mapbox-wagyu/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get, apply_conandata_patches, export_conandata_patches
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.52.0"
+
```
Comment (jellespijker) [inline recipes/mapbox-wagyu/all/conanfile.py:10]:
```suggestion


```

**Review by jellespijker — APPROVED** [review]

**Pull request merged**